# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webrequest/SecurityInfo.jsm
# Commit: 11e5baee978e
# Full Hash: 11e5baee978e8f98fa3795e430dd60c7600c9e8c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1580315
# File Overlap Count: 1
# Description:
#   Bug 1580315 - Convert certList to raw array for nsITransportSecurityInfo r=keeler,Ehsan,kershaw
#   
#   This patch converts the certList attribute of nsITransportSecurityInfo
#   from nsIX509CertList to Array<nsIx509Cert>
#   
# ==============================================================================

diff -r 341d72ef4bf5 -r 11e5baee978e toolkit/components/extensions/webrequest/SecurityInfo.jsm
--- a/toolkit/components/extensions/webrequest/SecurityInfo.jsm	Mon Oct 21 19:33:02 2019 +0000
+++ b/toolkit/components/extensions/webrequest/SecurityInfo.jsm	Mon Oct 21 19:49:01 2019 +0000
@@ -205,7 +205,7 @@
 
   getCertificateChain(certChain, options = {}) {
     let certificates = [];
-    for (let cert of certChain.getEnumerator()) {
+    for (let cert of certChain) {
       certificates.push(this.parseCertificateInfo(cert, options));
     }
     return certificates;